### PR TITLE
Logger

### PR DESCRIPTION
**Pull request by j0code**
- yes
- color
- timestamp
- yello

### CODE REVIEW

File: src/Logger.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+const GRAY  = ESC + "[37m"
+const WHITE = ESC + "[97m"
+const DARK  = ESC + "[90m"
+const RED   = ESC + "[91m"
+
+export default class Logger {
+
+	constructor(private readonly actor: string) {}
+	
+	log(message: string) {
```
Comment (cfpwastaken) [inline src/Logger.ts:12]: Make it log(...message: string)
And then make it log all of the args, like with console.log

File: src/Logger.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		console.log(`${WHITE}[${DARK}${time()}${WHITE}] [${GRAY}${this.actor}${WHITE}]${GRAY}`, message)
+	}
+
+	error(message: string) {
```
Comment (cfpwastaken) [inline src/Logger.ts:16]: Same here

File: src/Logger.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+		console.log(`${WHITE}[${DARK}${time()}${WHITE}] [${RED}${this.actor}${WHITE}]${RED}`, message)
+	}
+
```
Comment (cfpwastaken) [inline src/Logger.ts:19]: What about a warn() and debug() function like you told me on Discord?

File: src/Logger.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+const ESC = "\u001b"
+const RESET = ESC + "[m"
+const GRAY  = ESC + "[37m"
+const WHITE = ESC + "[97m"
+const DARK  = ESC + "[90m"
+const RED   = ESC + "[91m"
+
+export default class Logger {
+
+	constructor(private readonly actor: string) {}
```
Comment (cfpwastaken) [inline src/Logger.ts:10]: Needs a color so we can diff what "actor" it is using colors, like you told me on discord.

File: src/logger.ts
Comment (cfpwastaken) [inline src/logger.ts:1]: Delete this file

**Review by cfpwastaken — CHANGES_REQUESTED** [review]

File: src/Logger.ts (hunk 1 above)
Comment (j0code) [reply src/Logger.ts:12]: That breaks the coloring
Unless I do `message.join(" ")` of course
Should I do that?

Comment (cfpwastaken) [reply src/Logger.ts:12]: yes

File: src/logger.ts
Comment (j0code) [reply src/logger.ts:1]: I already have...

File: src/Logger.ts (hunk 3 above)
Comment (j0code) [reply src/Logger.ts:46]: addressed by: [df3c529](https://github.com/creelonestudios/mailverse/pull/9/commits/df3c529f72307e04790a33966149ff4d42d04a79)

File: src/Logger.ts (hunk 4 above)
Comment (j0code) [reply src/Logger.ts]: addressed by: [df3c529](https://github.com/creelonestudios/mailverse/pull/9/commits/df3c529f72307e04790a33966149ff4d42d04a79)

File: src/Logger.ts (hunk 1 above)
Comment (j0code) [reply src/Logger.ts]: addressed by: [2e81dde](https://github.com/creelonestudios/mailverse/pull/9/commits/2e81dde9c7433f77df3c5266410145f61422eee4)

File: src/logger.ts
Comment (j0code) [reply src/logger.ts:1]: addressed by: [2e81dde](https://github.com/creelonestudios/mailverse/pull/9/commits/2e81dde9c7433f77df3c5266410145f61422eee4)

File: src/Logger.ts (hunk 2 above)
Comment (j0code) [reply src/Logger.ts]: addressed by: [2e81dde](https://github.com/creelonestudios/mailverse/pull/9/commits/2e81dde9c7433f77df3c5266410145f61422eee4)

File: src/Logger.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import getConfig from "./config.js"
+
+const ESC = "\u001b"
+const RESET = ESC + "[m"
+const GRAY  = ESC + "[37m"
+const WHITE = ESC + "[97m"
+const DARK  = ESC + "[90m"
+const RED   = ESC + "[91m"
+const GREEN = ESC + "[92m"
+const YELLO = ESC + "[93m"
```
Comment (cfpwastaken) [inline src/Logger.ts:10]:
```suggestion
const YELLOW = ESC + "[93m"
```

File: src/Logger.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+const PINK  = ESC + "[95m"
+const TEAL  = ESC + "[96m"
+const WHITE = ESC + "[97m"
+
+const COLOR = {
+	GRAY, DARK, RED, GREEN, YELLO, PINK, TEAL, WHITE
```
Comment (cfpwastaken) [inline src/Logger.ts:16]:
```suggestion
	GRAY, DARK, RED, GREEN, YELLOW, PINK, TEAL, WHITE
```

File: src/Logger.ts [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+	log(...message: string[]) {
+		console.log(`${WHITE}[${DARK}${time()}${WHITE}] [${COLOR[this.color]}${this.actor}${WHITE}]${GRAY}`, message.join(" "))
+	}
+
+	error(...message: string[]) {
+		console.log(`${WHITE}[${DARK}${time()}${WHITE}] [${COLOR[this.color]}${this.actor}${WHITE}]${RED}`, message.join(" "))
+	}
+
+	warn(...message: string[]) {
+		console.warn(`${WHITE}[${DARK}${time()}${WHITE}] [${COLOR[this.color]}${this.actor}${WHITE}]${YELLO}`, message.join(" "))
```
Comment (cfpwastaken) [inline src/Logger.ts:36]:
```suggestion
		console.warn(`${WHITE}[${DARK}${time()}${WHITE}] [${COLOR[this.color]}${this.actor}${WHITE}]${YELLOW}`, message.join(" "))
```

File: src/pop3/POP3Server.ts [hunk 8]
```diff
@@ -4,6 +4,9 @@ import User from "../models/User.js"
 import { createHash } from "node:crypto"
 import { readFile } from "fs/promises"
 import Mail from "../models/Mail.js"
+import Logger from "../Logger.js"
+
+const logger = new Logger("POP3", "YELLO")
```
Comment (cfpwastaken) [inline src/pop3/POP3Server.ts:9]:
```suggestion
const logger = new Logger("POP3", "YELLOW")
```

**Review by cfpwastaken — CHANGES_REQUESTED** [review]
Rest is neis

File: src/pop3/POP3Server.ts (hunk 8 above)
Comment (j0code) [reply src/pop3/POP3Server.ts:9]: see above

File: src/Logger.ts (hunk 7 above)
Comment (cfpwastaken) [inline src/Logger.ts:36]:
```suggestion
		console.warn(`${WHITE}[${DARK}${time()}${WHITE}] [${COLOR[this.color]}${this.actor}${WHITE}]${YELLOW}`, message.join(" "))
```

File: src/pop3/POP3Server.ts (hunk 8 above)
Comment (cfpwastaken) [inline src/pop3/POP3Server.ts:9]:
```suggestion
const logger = new Logger("POP3", "YELLOW")
```

**Review by cfpwastaken — CHANGES_REQUESTED** [review]
After careful consideration, I have decided to reject your rejection.
I have to remind you that "Yellow" is actually the correct spelling of the color. See [here](https://en.wikipedia.org/wiki/Yellow).

**Comment by cfpwastaken** [comment]
Nix da

**Review by crystallic-j0code — APPROVED** [review]
I support the keepings of yello.

**Review by cfpwastaken — CHANGES_REQUESTED** [review]
.

**Review by crystallic-j0code — APPROVED** [review]

**Review by cfpwastaken — DISMISSED** [review]
.

**Review by cfpwastaken — APPROVED** [review]
Your stupid spelling mistake is safe.... For now.

**Review by cfpwastaken — APPROVED** [review]
Very gut

**Pull request merged**